integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/service/TenderApiService.java

```diff
@@ -161,38 +172,88 @@ public TenderApiService(
   }
 
   @Override
-  public ResponseEntity<AbandonITT200Response> invalidITT(AbandonITT abandonITT){
-
-    AbandonITT200Response response = new AbandonITT200Response();
+  public ResponseEntity<String> terminateRFx(String procID, String eventID) {
+
+    logger.info("Invalidating ITT  - ProcID : {} / EventID : {}", procID, eventID);
+
+    RfxGetResponse rfx =
+        rfxApi
+            .getRFX(String.format("rfxReferenceCode==%s", eventID))
+            .block(Duration.ofSeconds(defaultApiTimeout));
+
+    final RfxSetting rfxSetting;
+    if (rfx != null) {
+      if (rfx.getDataList() != null) {
+        if (rfx.getDataList().getRfx() != null) {
+          rfxSetting =
+              rfx.getDataList() != null ? rfx.getDataList().getRfx().get(0).getRfxSetting() : null;
+        } else {
+          throw new IllegalStateException(
+              String.format("No RFx details found for refReferenceCode - %s", eventID));
+        }
+      } else {
+        throw new IllegalStateException(
+            String.format("No RFx dataList found for refReferenceCode - %s", eventID));
+      }
+    } else {
+      throw new IllegalStateException(
+          String.format("No RFx found for refReferenceCode - %s", eventID));
+    }
 
-    logger.info("Invalidating ITT : {}", abandonITT);
+    Integer ownerUserId;
+    if (rfxSetting != null) {
+      ownerUserId = rfxSetting.getOwnerUser().getId();
+    } else {
+      throw new IllegalStateException(
+          String.format("No Owner found for refReferenceCode - %s", eventID));
+    }
 
+    String status = rfxSetting.getStatus();
     OperatorUser operatorUser = new OperatorUser();
-    operatorUser.setId(defaultOwnerUserId);
+    operatorUser.setId(ownerUserId);
 
     RfxInvalidate rfxInvalidate = new RfxInvalidate();
-    rfxInvalidate.setRfxReferenceCode(abandonITT.getRfxReferenceCode());
     rfxInvalidate.setOperatorUser(operatorUser);
+    rfxInvalidate.setRfxReferenceCode(eventID);
+
+    RfxInvalidationResponse jaggaerResponse;
+
+    if (status != null && status.equals("To be Published")) {
+      jaggaerResponse =
+          rfxWorkflowsApi.deleteRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    } else {
+      jaggaerResponse =
+          rfxWorkflowsApi.invalidateRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    }
+    final String finalStatus = (jaggaerResponse == null) ? "" : jaggaerResponse.getFinalStatus();
+
+    String response;
 
-    RfxInvalidationResponse jaggaerResponse = rfxWorkflowsApi.invalidateRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    // Convert Jaggaer termination status to OCDS
+    if (finalStatus != null && finalStatus.equalsIgnoreCase("Invalidated")) {
```

Comment:
Better as constants.

Reply:
I thought that only this morning when I went looking for them to answer a query from Nick